src/puppetlabs/dujour/version_check.clj

```diff
@@ -145,28 +160,55 @@
     (let [bytes (.digest md)]
       (reduce #(str %1 (format "%02x" %2)) "" bytes))))
 
+(defn- update-with-ids
+  [parameters]
+  (let [keys-present-to-hash (vec (keys (select-keys parameters [:cacert :certname])))]
+    (-> (ks/mapvals get-hash keys-present-to-hash parameters)
+        (set/rename-keys {:certname :host-id
+                          :cacert :site-id}))))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
-(defn check-for-updates!
+(defn ^:deprecated check-for-updates!
+  "Check whether a newer version of the software is available and submit
+  telemetry data. This is deprecated in favor of `check-for-update` and
+  `send-telemetry` which separate these two operations (since users may want to
+  opt out of telemetry while still being notified of new versions)."
   ([request-values update-server-url]
     (check-for-updates! request-values update-server-url nil))
   ([request-values update-server-url callback-fn]
     (validate-config! request-values update-server-url)
     (future
-      (let [keys-present-to-hash (vec (keys (select-keys request-values [:cacert :certname])))
-            arguments (-> (ks/mapvals get-hash keys-present-to-hash request-values)
-                          (set/rename-keys {:certname :host-id
-                                            :cacert :site-id}))
-            server-response (try
-                              (version-check arguments update-server-url)
-                              (catch Exception e
-                                (log/warn e "Error occurred while checking for updates")
-                                (throw e)))]
+      (let [arguments (update-with-ids request-values)
+            server-response (version-check arguments update-server-url)]
         (if-not (nil? callback-fn)
           (callback-fn server-response)
           server-response)))))
 
+(schema/defn check-for-update
+  "Check whether a newer version of the software is available. It does not
+  submit any extra data beyond the product name and version. If a newer version
+  is available it will log a message to that effect and return a map describing
+  the newer version (see `UpdateInfo`). If it is unable to connect to the server
+  it will log and return nil. "
```

Comment:
Yeah I'm for that. We could raise something like `kind: puppetlabs.version-check/request-failure` or something and the client can look for that and log what's appropriate.